src/main/java/io/github/dsheirer/record/wave/WaveWriter.java

```diff
@@ -155,7 +156,8 @@ private void rename() throws IOException
             {
                 String renamedWithSuffix = mFile.getFileName().toString();
                 renamedWithSuffix = renamedWithSuffix.replace(".tmp", "_" + suffix + ".wav");
-                Path renamedWithSuffixPath = mFile.resolveSibling(renamedWithSuffix);
+                String cleaned = StringUtils.replaceIllegalCharacters(renamedWithSuffix);
```

Comment:
This change can be removed.  The original filename should have valid characters if it got to this point, and so the updated suffix version should also have valid filename charactes.  Otherwise, this will corrupt linux filepaths.